samples/src/main/scala/org/llm4s/samples/embeddingsupport/EmbeddingExample.scala

```diff
@@ -18,27 +18,27 @@ object EmbeddingExample {
   private val logger = LoggerFactory.getLogger(getClass)
 
   // ------------------------ Output knobs (env-tunable) ------------------------
-  private val MAX_ROWS_PER_FILE: Int =
-    sys.env.get("MAX_ROWS_PER_FILE").flatMap(s => Try(s.toInt).toOption).getOrElse(200)
-  private val TOP_DIMS_PER_ROW: Int =
-    sys.env.get("TOP_DIMS_PER_ROW").flatMap(s => Try(s.toInt).toOption).getOrElse(6)
-  private val GLOBAL_TOPK: Int =
-    sys.env.get("GLOBAL_TOPK").flatMap(s => Try(s.toInt).toOption).getOrElse(10)
-  private val SHOW_GLOBAL_TOP: Boolean =
-    sys.env.get("SHOW_GLOBAL_TOP").exists(_.trim.equalsIgnoreCase("true"))
+  private def MAX_ROWS_PER_FILE(config: ConfigReader): Int =
+    config.get("MAX_ROWS_PER_FILE").flatMap(s => Try(s.toInt).toOption).getOrElse(200)
+  private def TOP_DIMS_PER_ROW(config: ConfigReader): Int =
+    config.get("TOP_DIMS_PER_ROW").flatMap(s => Try(s.toInt).toOption).getOrElse(6)
```

Comment:
This is a bit ugly / repetative.

I think we should have helper methods 

config.getOptionalInt("TOP_DIMS_PER_ROW", 6)